### PR TITLE
set v1.14 as the next release in azure-dev and test/metadata

### PR DESCRIPTION
**Pull request by nawazkh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- set v1.14 as the next release in azure-dev and test/metadata

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/data/shared/v1beta1_provider/metadata.yaml [hunk 1]
```diff
@@ -7,12 +7,12 @@
 # one capz version at a time against one API version
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 releaseSeries:
-  - major: 1
```
Comment (CecileRobertMichon) [inline test/e2e/data/shared/v1beta1_provider/metadata.yaml:10]: I think we need to keep this since we're still testing upgrade with it

Comment (nawazkh) [reply test/e2e/data/shared/v1beta1_provider/metadata.yaml:10]: Shall I add back 1.11 or update the base version we upgrade from in [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4473/files#diff-35f348a5d567ebeba39b1e40bbab0e1a7f64009cbadff142952f9b9691e94666L86)?

Comment (nawazkh) [reply test/e2e/data/shared/v1beta1_provider/metadata.yaml:10]: From ad-hoc discussion : will add back 1.11 to `test/e2e/data/shared/v1beta1_provider/metadata.yaml`

Comment (nawazkh) [reply test/e2e/data/shared/v1beta1_provider/metadata.yaml]: Added back release 1.11.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**